### PR TITLE
subsamplers: implement first video clipping subsampler

### PR DESCRIPTION
**Pull request by iejMac**

### CODE REVIEW

**Comment by iejMac** [comment]
https://github.com/iejMac/video2dataset/issues/50

File: video2dataset/worker.py [hunk 1]
```diff
@@ -48,7 +49,7 @@ def __init__(
         self.oom_shard_count = oom_shard_count
         self.encode_format = encode_format
         self.data_reader = VideoDataReader(video_height, video_width, timeout, find_format_timeout, max_format_tries)
-        self.subsampler = NoOpSubsampler()
+        self.clipping_subsampler = ClippingSubsampler(oom_clip_count)
```
Comment (rom1504) [inline video2dataset/worker.py:52]: I suggest we branch here to use either no op either clipping (and in the future others)

File: video2dataset/worker.py [hunk 2]
```diff
@@ -136,10 +138,11 @@ def data_generator():
                     )
                     continue
 
-                subsampled_video, error_message = self.subsampler(vid_stream)
+                subsampled_videos, metas, error_message = self.clipping_subsampler(vid_stream, meta)
```
Comment (rom1504) [inline video2dataset/worker.py:141]: same here, let's use no op or clipping subsampler depending on a param

**Comment by iejMac** [comment]
https://github.com/kkroening/ffmpeg-python/issues/266

**Comment by iejMac** [comment]
use segment_times to pass in split points and then based on clips spans choose the right segments

**Comment by iejMac** [comment]
tests: we want to check if the clips that are written are actually the correct ones. We can do this by comparing the times of the written mp4's and the differences in the span argument.

TODO's:
- [ ] read in a video + metadata containing clips arg
- [ ] use ClippingSubsampler on video_bytes, metadata and retrieve bytes of video fragments and metadata fragments
- [ ] compare, maybe using ffprobe + our get_seconds func

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@ jobs:
           source .env/bin/activate
           python -m pip install -U pip
           make install-dev
+          sudo apt-get install -y ffmpeg
```
Comment (iejMac) [inline .github/workflows/ci.yml:26]: note sure if this is the best way to do this

**Comment by iejMac** [comment]
maybe this is how https://github.com/microsoft/XPretrain/blob/42b75a95d5e3959b54b798f82782c0c96576df0a/hd-vila-100m/src/cut_videos.py#L56

**Comment by iejMac** [comment]
https://github.com/RyanMarten/distributed_gcp_youtube_download/blob/07a1a797b46eb073ee632f78cf8716826caf002f/hdvila10m/download_hdvila10m.sh#L22 force keyframes at cuts?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (rom1504) [reply .github/workflows/ci.yml:26]: Is it not included when pip installing?

File: requirements-test.txt [hunk 4]
```diff
@@ -4,4 +4,5 @@ pylint==2.13.4
 pytest-cov==3.0.0
 pytest-xdist==2.5.0
 pytest==7.0.1
-types-requests
\ No newline at end of file
+types-requests
+ffprobe-python
```
Comment (rom1504) [inline requirements-test.txt:8]: No ffmpeg ?

**Pull request merged**